code/modules/mob/inventory.dm

```diff
@@ -239,10 +239,8 @@ var/list/slot_equipment_priority = list(
 	return TRUE
 
 //This function is an unsafe proc used to prepare an item for being moved to a slot, or from a mob to a container
-//It should be equipped to a new slot or forcemoved somewhere immediately after this is called
+//It should be equipped to a new slot or forcemoved somewhere immediately after this is calleds
 /mob/proc/prepare_for_slotmove(obj/item/I)
-	if(!canUnEquip(I))
-		return 0
```

Comment:
there was no need for this check to be here. Prepare for slotmove should only be called after these checks are already done for the current stack